bin/exercise.ss

```diff
@@ -10,15 +10,17 @@
                     (output . ,output))))))
 
 (define (report-results chez guile)
-  (case (+ (length chez) (length guile))
-    (0 (output-results "fail" '() "Syntax error"))
-    (else 
-      (let ((choice (if (<= (failure-count chez) (failure-count guile))
-                      chez
-                      guile)))
+  (if (zero? (+ (length chez) (length guile)))
+    (output-results "fail" '() "Syntax error")
+    (let ((choice (cond
+                    ((null? guile) chez)
+                    ((null? chez) guile)
+                    (else (if (<= (failure-count chez) (failure-count guile))
+                            chez
+                            guile)))))
       (if (zero? (failure-count choice))
           (output-results "pass" (failure-messages choice) "")
-          (output-results "fail" (failure-messages choice) ""))))))
+          (output-results "fail" (failure-messages choice) "")))))
```

Comment:
Good catch, @austinmlv .  We should have been handling the case of an empty list.  I actually think the procedure `report-results` could stand to have deeper refactoring.  What do you think about this?

``` scheme
(define report-results
  (let ((pass-or-fail
         (lambda (result)
           (output-results
            (or (and (zero? (failure-count result)) "pass") "fail")
            (failure-messages result) ""))))
    (lambda (chez guile)
      (cond
       ((andmap null? `(,chez ,guile))
        (output-results "fail" '() "Syntax error"))
       (else
        (if (not (null? chez)) (pass-or-fail chez)
            (pass-or-fail guile)))))))
```

Reply:
That looks good. The use of `andmap` really helps to clean things up.
I incorporated the changes into a new commit dda5e16 but I think I screwed something up when rebasing onto main.

Reply:
Okay this PR should be good to go now.

Reply:
Well, I won't try to force you into anything you don't want to do.  Fwiw, the main branches are protected, and screw ups are almost always reversible.  The boon to having you as a maintainer is we could approve each other's PRs, so we wouldn't need to get someone like Eric, who is not normally thinking about scheme, to review the work.